app/models/host/managed.rb

```diff
@@ -167,6 +167,10 @@ def compute_provides?(attr)
     end
   end
 
+  def fqdn_options
+    facts['fqdn'] || name
```

Comment:
I think we should prefer `name` over facts when unattended is true, as it should always be a copy of the FQDN from the NIC.